include/sourcepp/detail/StringUtils.h

```diff
@@ -22,4 +25,195 @@ void trim(std::string& s, std::string_view c);
 
 std::vector<std::string> split(std::string_view s, char delim);
 
+class UtilStringView
+{
+    std::string owningString;
+    std::string_view string;
+    bool selfOwning = false;
+public:
+
+    UtilStringView(const char* str)
+    {
+        owningString = str;
+        string = std::string_view(owningString);
+        selfOwning = true;
+    }
+
+    UtilStringView(std::string& str)
+    {
+        owningString = std::move(str);
+        string = std::string_view(owningString);
+        selfOwning = true;
+    }
+
+    UtilStringView(std::string_view str)
+    {
+        string = str;
+    }
+
+    UtilStringView() = default;
+
+    ~UtilStringView() = default;
+
+    template<typename T>
+    explicit UtilStringView(const UtilStringView& t){
```

Comment:
Rule of 3/5/0: if you implement copy and move constructors, you must implement copy and move assignment operators as well